api/services/User/index.js

```diff
@@ -39,47 +39,57 @@ export default class User extends BaseModelService {
   }
 
   static async findAllAvailabilityDuring(keys) {
-    return Promise.all(
-      keys.map(
-        async (key) =>
-          !(await Models.CalendarEventInvite.scope('visible').findOne({
+    return Models.CalendarEventInvite.scope('visible')
+      .findAll({
+        where: {
+          user_id: {
+            [Models.Sequelize.Op.in]: keys.map(({ user_id }) => user_id),
+          },
+        },
+        include: [
+          {
+            model: Models.CalendarEvent,
+            as: 'event',
             where: {
-              user_id: key.user_id,
-            },
-            include: [
-              {
-                model: Models.CalendarEvent,
-                as: 'event',
-                where: {
+              [Models.Sequelize.Op.or]: keys.map(
+                ({ user_id, dateTimeRange, excludeEvents }) => ({
                   [Models.Sequelize.Op.and]: [
-                    overlapDateTimeClause(key.dateTimeRange),
+                    {
+                      $user_id$: user_id,
+                    },
                     {
                       id: {
-                        [Models.Sequelize.Op.notIn]: key.exculdeEvents || [],
+                        [Models.Sequelize.Op.notIn]: excludeEvents || [],
                       },
                     },
+                    overlapDateTimeClause(dateTimeRange),
                   ],
-                },
-                required: true,
-              },
-            ],
-          })),
-      ),
-    );
+                }),
+              ),
+            },
+            required: true,
+          },
+        ],
+      })
+      .then((invites) =>
+        keys.map(
+          (key) => !invites.find((invite) => invite.user_id === key.user_id),
```

Comment:
I don't think we need to deep equate here.
What we want to do is, to make sure that the `invite` is corresponding to this `key`, not to some other key. Right?. So we will have to check if invite timing are between `key.dateTimeRange`. ( similar to `overlapDateTimeClause(dateTimeRange)` 
But because we are handling that inside the `where` query of `include`, that will always be true. if `user_id` of invite and key are same then `key.DateTimeRange` must be between `invite.start_at and invite.end_at`

Am i making any sense? :sweat_smile:

Reply:
What if there are keys with save user_id but different other filters?

Also the current query structure has to be changed. Check the previous comment.

Reply:
I have changed the query

Reply:
No please re-read my first comment